src/core/style/expressions/torque.js

```diff
@@ -0,0 +1,103 @@
+import Expression from './expression';
+import { implicitCast, DEFAULT } from './utils';
+import { floatDiv, floatMod, now, linear, globalMin, globalMax } from '../functions';
+import Property from './property';
+
+const DEFAULT_FADE = 0.15;
+
+/**
+ * Create a Torque FadeIn/FadeOut configuration
+ *
+ * @param {carto.style.expression.expression|Number} param1 expression of type float or Number
+ * @param {carto.style.expression.expression|Number} param2 expression of type float or Number
+ * @return {carto.style.expressions.fade}
+ *
+ * @example `
+// fadeIn of 0.1 seconds, fadeOut of 0.3 seconds
+filter:       torque($day, 40, fade(0.1, 0.3))
+
+// fadeIn and fadeOut of 0.5 seconds
+filter:       torque($day, 40, fade(0.5))
+
+`
+ *
+ * @memberof carto.style.expressions
+ * @name fade
+ * @function
+*/
+export class Fade extends Expression {
+    constructor(param1 = DEFAULT, param2 = DEFAULT) {
+        let fadeIn = param1;
+        let fadeOut = param2;
+        if (param1 == DEFAULT) {
+            fadeIn = DEFAULT_FADE;
+        }
+        if (param2 == DEFAULT) {
+            fadeOut = fadeIn;
+        }
+        fadeIn = implicitCast(fadeIn);
+        fadeOut = implicitCast(fadeOut);
+        super({ fadeIn, fadeOut });
+        this.type = 'fade';
+        this.inlineMaker = (inline) => ({
+            in: inline.fadeIn,
+            out: inline.fadeOut,
+        });
+    }
+}
+
+/**
+ * Create an animated temporal filter (torque)
+ *
+ * @param {carto.style.expression.expression} input input to base the temporal filter,
+ * if input is a property, the beginning and end of the animation will be determined by the minimum and maximum timestamps of the property on the dataset,
+ * this can be problematic if outliers are present. Otherwise input must be a float expression in which 0 means beginning of the animation and 1 means end.
+ * @param {Number} duration duration of the animation in seconds
+ * @param {carto.style.expression.fade} fade fadeIn/fadeOut configuration
+ * @return {carto.style.expressions.torque}
+ *
+ * @example `width:    2
```

Comment:
🌈  wrong indentation

Reply:
In the example? How should it be?

Reply:
- keep the `*` at the beginning of the lines
- the example starts in the next line (after @example tag) 
- there is no need for the backtick
-

Reply:
```javascript
 * @example <caption> fadeIn of 0.1 seconds, fadeOut of 0.3 seconds </caption>
 * const s = carto.style.expressions;
 * new carto.Style({
 *  filter: s.torque($day, 40, s.fade(0.1, 0.3))
 * });
 * 
 * @example <caption>   fadeIn and fadeOut of 0.5 seconds </caption>
 * const s = carto.style.expressions;
 * new carto.Style({
 *  filter: s.torque($day, 40, s.fade(0.5))
 * });
```

Reply:
:+1: Thank you, updated